### ISSUE TITLE
datagen.standardize(x) alters original x

### ISSUE THREAD

**Issue by trane293** [fchollet/keras#4724]
Hello, recently I have noticed this issue that when I try to standardize my test set using my training data generator (let's call it "datagen"), using the line:

`X_TEST_scaled = datagen.standardize(X_TEST)`

It returns the scaled version of X_TEST, but it also alters the original X_TEST tensor, which makes me lose my original X_TEST data. 

Please let me know if I am doing something wrong, or is this the intended behaviour (in which it's pretty weird).

**Comment by erilyth** [comment]
@trane293 You would need to create a copy. Take a look at this http://stackoverflow.com/questions/3059395/numpy-array-assignment-problem

Solution: X_TEST_scaled = datagen.standardize(X_TEST.copy())

**Comment by trane293** [comment]
Of course I can create a copy, that's what I've been doing to counter this issue. But I wanted to know if this is the intended behaviour or a bug with implementation? Traditionally functions don't modify arguments, and return new variables while preserving the arguments. This doesn't seem to be the case here.

**Comment by erilyth** [comment]
I don't think its an issue, numpy arrays (objects) do this and its just how they work.
